### ISSUE TITLE
ServerBag assumes the use of PHP_AUTH_USER denotes Basic authentication

### ISSUE THREAD

**Issue by cjmaio** [symfony/symfony#17345]
I use the Laravel framework, which makes use of `symfony/http-foundation`.

We developed an API, and recently moved to using a Shibboleth IdP. When we send through our header to Laravel, `Authorization: Bearer 123456.....` it was being stripped and replaced with `Authorization: Basic 123456.....`.

I couldn't determine why it was being stripped. After hours of thinking it was my own code, I finally tracked it down to line 89 of `symfony/http-foundation/ServerBag.php`:

```
// PHP_AUTH_USER/PHP_AUTH_PW
if (isset($headers['PHP_AUTH_USER'])) {
    $headers['AUTHORIZATION'] = 'Basic '.base64_encode($headers['PHP_AUTH_USER'].':'.$headers['PHP_AUTH_PW']);
} elseif (isset($headers['PHP_AUTH_DIGEST'])) {
    $headers['AUTHORIZATION'] = $headers['PHP_AUTH_DIGEST'];
}
```

It seems that this is assuming that if `PHP_AUTH_USER` is set, that it is Basic authentication. However, the Apache module `mod_shib` uses `PHP_AUTH_USER` when a user is logged in using the IdP.

Should this be instead something like this?

```
// PHP_AUTH_USER/PHP_AUTH_PW
if (isset($headers['PHP_AUTH_USER'])) {
    if (isset($headers['AUTH_TYPE']) && $headers['AUTH_TYPE'] == 'Basic') {
        $headers['AUTHORIZATION'] = 'Basic '.base64_encode($headers['PHP_AUTH_USER'].':'.$headers['PHP_AUTH_PW']);
    }
} elseif (isset($headers['PHP_AUTH_DIGEST'])) {
    $headers['AUTHORIZATION'] = $headers['PHP_AUTH_DIGEST'];
}
```

**Comment by jakzal** [comment]
`PHP_AUTH_USER` is set based on the `AUTHORIZATION` header, only for basic and digest authentication methods. How's that possible you've got it set for a bearer authentication?

**Comment by cjmaio** [comment]
We are using JWT for keeping logged in on a mobile application, using Laravel (and therefore this Symfony module) as the API. Our servers however use Shibboleth / mod_shib for authentication, which sets PHP_AUTH_USER.

**Comment by jakzal** [comment]
@cjmaio it doesn't sound right. Could you point me to the Shibboleth documentation that would help me understand why they are setting `PHP_AUTH_USER`?

**Comment by cjmaio** [comment]
Unfortunately I can't seem to find any documentation specifically saying that it sets `PHP_AUTH_USER`. However, I did setup a simple test on our server with a directory requiring Shibboleth authentication with the following `.htaccess` file:

```
AuthType shibboleth
ShibRequireSession On
Require valid-user
```

Once logging in and being able to access the simple `index.php` I added to that directory...

```
<?php var_dump($_SERVER); ?>
```

It outputs `REMOTE_USER` and `PHP_AUTH_USER` as my currently logged in username (cjmaio@uwm.edu).

Our shibboleth configuration is pretty out of the box. For reference, we are using RHEL 7.1 with Apache 2.4 and PHP 5.6. I'm wondering if PHP is perhaps setting the `PHP_AUTH_USER` variable to what `REMOTE_USER` is set to, because of the .htaccess file. I know that mod_shib **does** set the `REMOTE_USER` variable.
